crates/uv-python/src/lib.rs

```diff
@@ -2603,4 +2693,45 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn find_python_pyodide() -> Result<()> {
+        let mut context = TestContext::new()?;
+
+        context.add_pyodide_version("3.13.2")?;
+        let python = context.run(|| {
+            find_python_installation(
+                &PythonRequest::Default,
+                EnvironmentPreference::Any,
+                PythonPreference::OnlySystem,
+                &context.cache,
+                Preview::default(),
+            )
+        })??;
+        // We should find the Pyodide intepreter
+        assert_eq!(
+            python.interpreter().python_full_version().to_string(),
+            "3.13.2"
+        );
+
+        // We should prefer any native Python to the Pyodide Python
+        context.add_python_versions(&["3.15.7"])?;
+
+        let python = context.run(|| {
+            find_python_installation(
+                &PythonRequest::Default,
+                EnvironmentPreference::Any,
+                PythonPreference::OnlySystem,
+                &context.cache,
+                Preview::default(),
+            )
+        })??;
+        // TODO: How to fix this?
+        assert_eq!(
+            python.interpreter().python_full_version().to_string(),
+            "3.15.7"
+        );
```

Comment:
@zanieb I tried to figure out how to sort the Emscripten interpreters after all the others but couldn't figure out how. Maybe `find_python_installation` should just discard them? I think it makes the most sense to only use them when specifically requested.

Reply:
Okay I did something slightly ad hoc...